src/main/bash/sdkman-use.sh

```diff
@@ -38,7 +38,10 @@ function __sdk_use() {
 	__sdkman_set_candidate_home "$candidate" "$version"
 
 	if [[ $PATH =~ ${SDKMAN_CANDIDATES_DIR}/${candidate}/([^/]+) ]]; then
-		export PATH=${PATH//${SDKMAN_CANDIDATES_DIR}\/${candidate}\/${BASH_REMATCH[1]}/${SDKMAN_CANDIDATES_DIR}\/${candidate}\/${version}}
+		local matched_version
+		[[ "$zsh_shell" == "true"  ]] && matched_version=${match[1]} || matched_version=${BASH_REMATCH[1]}
```

Comment:
Sorry to be a pain, but do you mind if we break this out into a proper `if` construct for consistency with the rest of the codebase?

Reply:
Yes, this makes sense.

Reply:
Done!